src/components/QuarantineSpeedDialAction/index.jsx

```diff
@@ -0,0 +1,76 @@
+import { PureComponent, Fragment } from 'react';
+import { func } from 'prop-types';
+import { format, addYears } from 'date-fns';
+import TextField from 'material-ui/TextField';
+import HomeLockIcon from 'mdi-react/HomeLockIcon';
+import HomeLockOpenIcon from 'mdi-react/HomeLockOpenIcon';
+import SpeedDialActionDialog from '../SpeedDialActionDialog';
+import { date } from '../../utils/prop-types';
+
+/**
+ * Display a speed dial action able to quarantine a worker.
+ * Clicking the button provides the ability to change the
+ * state (`quarantineUntil`) of the worker via a dialog.
+ */
+export default class QuarantineSpeedDialAction extends PureComponent {
+  static propTypes = {
+    /**
+     * Once the `quarantineUntil` time has elapsed, the worker
+     * resumes accepting jobs.
+     */
+    quarantineUntil: date,
+    /** The callback to trigger when the action button is clicked */
+    onQuarantineClick: func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      quarantineUntil: props.quarantineUntil
+        ? new Date(props.quarantineUntil)
+        : addYears(new Date(), 1000),
+    };
+  }
+
+  handleQuarantineClick = () => {
+    this.props.onQuarantineClick(format(this.state.quarantineUntil));
+  };
+
+  handleQuarantineChange = ({ target }) => {
+    this.setState({ quarantineUntil: new Date(target.value) });
+  };
+
+  render() {
+    const { quarantineUntil, onQuarantineClick: _, ...props } = this.props;
+
+    return (
+      <SpeedDialActionDialog
+        icon={quarantineUntil ? <HomeLockOpenIcon /> : <HomeLockIcon />}
+        tooltipTitle={quarantineUntil ? 'Update Quarantine' : 'Quarantine'}
+        title="Quarantine?"
+        body={
+          <Fragment>
+            <Fragment>
+              Quarantining a worker allows the machine to remain alive but not
+              accept jobs. Note that a quarantine can be lifted by setting
+              Quarantine Until to the present time or somewhere in the past.
+            </Fragment>
+            <br />
+            <br />
+            <TextField
```

Comment:
@eliperelman I tried using `material-ui-pickers` but having two modals open at the same time is not so pretty. Also, I want to point out that I've been having trouble making the `onChange` handler work on the `material-ui-pickers` component. I suspect it has something to do with the `z-index` since 2 modals are open. tbh, I think, for now, leaving the current basic implementation should be okay. We can revisit later.

![screen shot 2018-04-25 at 3 50 05 pm](https://user-images.githubusercontent.com/3766511/39269435-cc6b908c-48a0-11e8-98bb-3db48a3802d3.png)

Reply:
The current implementation is pretty ugly as well...ugh